### DISCUSSION TITLE
Can you pass PageData to a component for reading and writing to the database from a SvelteKit component?

### DISCUSSION THREAD

**Discussion by ethanfox** [supabase/supabase#14188] [Questions] [answered]
Can you pass PageData to a component for reading and writing to the database from a SvelteKit component?
Is it possible to pass

```
import type { PageData } from './$types';

export let data: PageData;
```

To a component so I can read, write, and update from a component rather than just a +page in Sveltekit?

I have nested components in order to make things easier to read/follow but there is only an example in the SvelteKit Auth Helpers involving a +page.svelte and no svelte component. Is this possible?

**Comment by danielbayley80** [comment]
Did you ever solve this?
I was wondering if it is directly accessible within components, whether to pass it like you suggest, or perhaps using a context. The same with ActionData

**Reply by ethanfox** [reply, marked as answer]
Yes look up setContext and getContext in the SvelteKit docs. This allows you to pass the page data to components.
